### ISSUE TITLE
Auto abandon channel opens for non broadcast-able funding tx

### ISSUE THREAD

**Issue by AbelLykens** [lightningnetwork/lnd#4503]
### Background

Was opening some channels. One of them is still in `lncli listpendingchannels`, but logs tell me: 

    lnd[31094]: 2020-08-03 07:26:56.827 [ERR] FNDG: Unable to rebroadcast funding tx 020000000001056***1d47157f84d1e36200000000 for ChannelPoint(6bd47e3ac6***:1): transaction rejected: output already spent

### Your environment

* LND v0.11.0-beta.rc1 (247b7530caf08a555ffd56f81019031bc1af6565)
* Linux 4.18.0-193.6.3.el8_2.x86_64 #1 SMP
* bitcoind bitcoin-0.20.0
* bitcoind running on separate machine

### Steps to reproduce

Open many channels. Set Spend Unconfirmed for all channels you open. As channels are being opened, some UTXOs will be spent which are meant for new channel opens. These remain in pending channels.

### Expected behaviour

I expect "impossible" opens (utxos spent) to disappear from pending channels.

### Actual behaviour

Channel remains in listpendingchannels.

**Comment by AbelLykens** [comment]
```    "pending_open_channels": [
        {
            "channel": {
                "remote_node_pub": "02e89ca9e8da72b33d896bae51d20e7e6675aa971f7557500b6591b15429e717f1",
                "channel_point": "6bd47e3ac613c2b6210a3d8b1262e0bfc6300ed7685125d2ebba85b706e2b055:1",
                "capacity": "10000000",
                "local_balance": "9987039",
                "remote_balance": "0",
                "local_chan_reserve_sat": "100000",
                "remote_chan_reserve_sat": "100000",
                "initiator": "INITIATOR_LOCAL",
                "commitment_type": "STATIC_REMOTE_KEY"
            },
            "confirmation_height": 0,
            "commit_fee": "12961",
            "commit_weight": "600",
            "fee_per_kw": "17903"
        }
    ],
```
pending channels output

**Comment by AbelLykens** [comment]
Just "fixed" (abandoned):
`lncli abandonchannel 6bd47e3ac613c2b6210a3d8b1262e0bfc6300ed7685125d2ebba85b706e2b055 1`
Thanks to hints by @carlaKC . 

Note for future readers: It's just the remote party still thinking this will be opened. According to Carla (if they use LND) the tx/open should be auto abandoned after 2w.

**Comment by carlaKC** [comment]
> the tx/open should be auto abandoned after 2w.

All implementations should forget about it after a reasonable [timeout](https://github.com/lightningnetwork/lightning-rfc/blob/master/02-peer-protocol.md#rationale-3), LND is 2016 blocks, I imagine the other implementations have something similair, just don't know them offhand. 

Thinking about the issue of pending channels with inputs that get spent elsewhere, we currently just wait for the funding tx to confirm when we are the initiator. In this case, the input UTXO was spent elsewhere, so there's no further action required other than nicer cleanup than manually having to abandon. 

This should be addressed as part of #386, which is on the board for 0.12, so I'm going to close this issue.
